04.md

```diff
@@ -14,7 +14,7 @@ A special `linkingKey` can be used to login user to a service or authorise sensi
 
 When creating an `LNURL-auth` handler `LN SERVICE` must include in it a `k1` query parameter consisting of randomly generated 32 bytes of data as well as optional `action` enum, an example is `https://site.com?tag=login&k1=hex(32 bytes of random data)&action=login`.
 
-Later, once `LN SERVICE` receives a call at the specified `LNURL-auth` handler, it should take `k1`, `key` and a DER-encoded `sig` and verify the signature using `secp256k1`. Once signature is successfully verified a user provided `key` can be used as identifier and may be stored in a session, database or however `LN SERVICE` sees fit.
+Later, once `LN SERVICE` receives a call at the specified `LNURL-auth` handler, it MUST take `k1`, compressed (65-byte) `secp256k1` public `key` encoded as hex, and a DER-encoded ECDSA `sig` and verify the signature using `secp256k1`. Once signature is successfully verified a user provided `key` can be used as identifier and may be stored in a session, database or however `LN SERVICE` sees fit.
```

Comment:
The uncompressed secp256k1 public key is 65 bytes. Compressed is 33 bytes.

Reply:
Can also specify that the `sig` should also be encoded as hex..

Reply:
Lol, good catches! Fixed.